include/deal.II/fe/mapping_fe.h

```diff
@@ -0,0 +1,602 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_mapping_fe_h
+#define dealii_mapping_fe_h
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/derivative_form.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/table.h>
+#include <deal.II/base/vectorization.h>
+
+#include <deal.II/fe/mapping.h>
+
+#include <deal.II/grid/tria_iterator.h>
+
+#include <deal.II/matrix_free/shape_info.h>
+
+#include <array>
+#include <cmath>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+/*!@addtogroup mapping */
+/*@{*/
+
+
+/**
+ * This class consistently uses a user-provided finite element on all cells of a
+ * triangulation to implement a polynomial mapping.
+ *
+ * If one initializes this class with an FE_Q object, then this class is
+ * equivalent to MappingQGeneric..Please not that no optimizations exploiting
+ * tensor-product structures of finite elements have been added here.
+ *
```

Comment:
Please add a sentence that if you choose the same FiniteElement as your discretization, that one would obtain an iso-parametric mapping.
Also say that using it with an FE_Q(degree) here has the same effect as MappingQ(degree).